students/RimleeT/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,120 @@
+"""	
+
+def do_thankyou():
+    print("<<<SEND A THANK YOU>>>")
+                
+    #if user types 'list' show the list of donors and prompt to enter donor name
+    donor_name = None
+    while True:
+        donor_name = input("Enter the name of donor or type 'list' if you wish to view the list of donors: ")
+        if donor_name.lower() == 'list':
+            print(donor_db)
+        else:
+            break
+    donation_amount = get_donation_amount() 
+
+    if donor_name not in donor_db:
+        donor_db[donor_name] = [donation_amount]
+    else:
+        donor_db[donor_name].append(donation_amount)
+    
+    print("donor_db",donor_db)
+    print (f"Thank you {donor_name} for your generous donation amount of ${donation_amount}" )
+
+def get_donation_amount():
+    try:
+        return float(input("Enter the donation amount: "))
+    except:
+        print("Invalid value. Type a number.")
+        return get_donation_amount()
+
+
+"""
+Create a Report
+If the user (you) selected “Create a Report,” print a list of your donors, sorted by total historical donation amount.
+Include Donor Name, total donated, number of donations, and average donation amount as values in each row. You do not need to print out all of each donor’s donations, just the summary info.
+Using string formatting, format the output rows as nicely as possible. The end result should be tabular (values in each column should align with those above and below).
+After printing this report, return to the original prompt.
+At any point, the user should be able to quit their current task and return to the original prompt.
+From the original prompt, the user should be able to quit the script cleanly.
+Your report should look something like this:
+
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+William Gates, III         $  653784.49           2  $   326892.24   
+Mark Zuckerberg            $   16396.10           3  $     5465.37
+Jeff Bezos                 $     877.33           1  $      877.33
+Paul Allen                 $     708.42           3  $      236.14
+"""
+
+def gen_stats(k,v):
+	donations = v
+	total_donation = float(format(sum(donations),'.2f'))
+	num_donation = len(donations)
+	avg_donation = float(format(total_donation/num_donation,'.2f'))
+	print("{0:18}      ${1:13}{2:15}     ${3:12}".format(k,total_donation, num_donation,avg_donation))
+
+def do_createreport():
+    print("<<<PRINTING REPORT>>>")
+    sorted_x = sorted(donor_db.items(), key=lambda item: sum(item[1]), reverse=True)
+    print("{0:23}|  {1:15}|  {2:12}|  {3:10}".format('Donor Name','Total Given','Num Gifts', 'Average Gifts'))
+    print('-' * 70)
+    for key,value in sorted_x :
+        gen_stats(key,value)
+    print('-' * 70)
+
+
+def do_sendlettertoalldonors():
+    dir_name = tempfile.mkdtemp(prefix = 'Letters-', dir = "C:/Users/rimle/SP_2019_210A_classroom/students/RimleeT/lesson04/")
```

Comment:
Be careful with setting a specific dir here, when I run this from my machine it throws an error. You can just create the files within this directory itself for now. (No need to specify the path)